assets/src/edit-story/components/panels/layer/useLayers.js

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import { useStory } from '../../../app';
+import { createNewElement } from '../../../elements';
+
+function getElementsWithBackground( page ) {
+	const hasBackground = Boolean( page.backgroundElementId );
+
+	// if no background element, create empty background element as first element.
+	if ( ! hasBackground ) {
+		return [
+			createNewElement( 'background' ),
```

Comment:
I think this looks a lot better. One thing I wanted to discuss, but don't have a very strong opinion on: what do you think if we made this the object shape for all element types? E.g. `Layer{type:string, inner:Element}`? This shape would be more similar and we'd only end up duplicating a single `type` string, but not in all cases. WDYT?

Reply:
Hm, I don't really see how that would work better. I need the icon and layer content based off the element/layer type, that's the main reason for this structure.

Can you explain your idea in more detail?

Reply:
I was more concerned that we may have to start transferring more and more parameters from  the `inner` to the `{type: background, ...}` top level. But that might be premature. Currently most of shared parameters (x/y/w/h/rotation/etc) are not really applicable to the background element. The concept of a layer is not exactly the same as an element, but I guess for now it's not really an issue.

Reply:
Yeah, it did require me to move the element types around a bit and didn't feel super clean. But I can't come up with a better solution currently.

Reply:
Ok. Let's move with this. I'm sure the right structure will pop up at some point.